src/Notifier.php

```diff
@@ -226,4 +236,22 @@ protected function getNoticesURL()
             $this->opt['projectKey']
         );
     }
+
+    protected function configureDefaultFilters()
+    {
+        $this->addFilter(function ($notice) {
+            if (!empty($this->opt['rootDirectory']) && !empty($notice['errors'])) {
+                $projectRoot = $this->opt['rootDirectory'];
+                foreach ($notice['errors'] as &$error) {
+                    if (empty($error['backtrace'])) {
+                        continue;
+                    }
+                    foreach ($error['backtrace'] as &$frame) {
+                        $frame['file'] = preg_replace("~^$projectRoot~", '[PROJECT_ROOT]', $frame['file']);
+                    }
+                }
+            }
+            return $notice;
+        });
+    }
```

Comment:
I suggest to create `rootDirectoryFilter`:

```
protected function rootDirectoryFilter()
{
    foreach ($notice['errors'] as &$error) {
        ...
    }
}
```

and then add it like this in constructor:

```
if (!empty($this->opt['rootDirectory']) {
    $this->addFilter($this->rootDirectoryFilter);
}
```

This way we can write a test for this filter.